### PR TITLE
JsonRpcProvider: Basic access authentication support

### PR DESCRIPTION
**Pull request by msmania**
With this patch, `JsonRpcProvider` accepts a url with basic authentication credentials like this:

```js
const PoktEndpoint = 'https://scott:tiger@pocket-node.example.com';
const provider = new JsonRpcProvider({
  rpcUrl: PoktEndpoint,
  dispatchers: [PoktEndpoint],
});
```

### CODE REVIEW

File: packages/provider/src/json-rpc-provider.ts [hunk 1]
```diff
@@ -93,14 +93,25 @@ export class JsonRpcProvider implements AbstractProvider {
...
 
-    const routedRpcUrl = `${finalRpcUrl}${route}`
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    const url = new URL(route, finalRpcUrl);
+    if (url.username || url.password) {
+      const credential =
+        Buffer.from(`${url.username}:${url.password}`).toString('base64');
```
Comment (kutoft) [inline packages/provider/src/json-rpc-provider.ts:104]: I'm not positive how this package is handling polyfills, so depending on that `Buffer` could potential cause problems.

**Review by kutoft — APPROVED** [review]
Everything looks good, just one comment.

**Comment by kutoft** [comment]
> With this patch, `JsonRpcProvider` accepts a url with basic authentication credentials like this:
> 
> ```js
> const PoktEndpoint = 'https://scott:tiger@pocket-node.example.com';
> const provider = new JsonRpcProvider({
>   rpcUrl: PoktEndpoint,
>   dispatchers: [PoktEndpoint],
> });
> ```

This could also be helpful in the READ.ME

File: packages/provider/src/json-rpc-provider.ts (hunk 1 above)
Comment (msmania) [reply packages/provider/src/json-rpc-provider.ts:104]: Ok, I imported the "buffer" module as another module (e.g. [signer](https://github.com/pokt-network/pocket-js/blob/main/packages/signer/src/key-manager.ts#L3)) in this repo does.

**Review by nodiesBlade — DISMISSED** [review]

**Review by nodiesBlade — APPROVED** [review]

**Pull request merged**